File: Java/src/main/java/QuestionBank/Solution401_500/Solution415.java
package QuestionBank.Solution401_500;

/**
 * @author windows
 */
public class Solution415 {
    public String addStrings(String num1, String num2) {
        int len1 = num1.length() - 1;
        int len2 = num2.length() - 1;
        int carry = 0;
        StringBuilder sb = new StringBuilder();
        while(len1 >= 0 || len2 >= 0){
            int a = len1 >= 0 ? num1.charAt(len1) - '0' : 0;
            int b = len2 >= 0 ? num2.charAt(len2) - '0' : 0;
            int sum = a + b + carry;
            sb.append(sum % 10);
            carry = sum / 10;
            len1--;
            len2--;
        }
        if (carry > 0){
            sb.append(carry);
        }
        return sb.reverse().toString();
    }


    public static void main(String[] args) {
        Solution415 solution415 = new Solution415();
    }
}

File: C++/src/Solution401_500/Solution415.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution415.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

string Solution415::addStrings(string num1, string num2) {
    int i = num1.length() - 1, j = num2.length() - 1;
    int carry = 0;
    string res = "";
    while (i >= 0 || j >= 0 || carry == 1){
        int sum = carry;
        if (i >= 0){
            sum += num1[i--] - '0';
        }
        if (j >= 0){
            sum += num2[j--] - '0';
        }
        carry = sum / 10;
        res = to_string(sum % 10) + res;
    }
    return res;
}
